### DISCUSSION TITLE
What is the correct way to add sharepoint shared drives with the docker installation? + is it worth it?

### DISCUSSION THREAD

**Discussion by kendonB** [abraunegg/onedrive#2345] [Q&A] [answered]
Hi,

The discussion here: https://github.com/abraunegg/onedrive/discussions/1996 suggests that one needs to find a new machine and install onedrive on that to even get the sharepoint details for use by the docker installation. Shouldn't that discourage anyone from even using docker? I would have to successfully install onedrive manually on the other machine anyway. What is the benefit of using docker when one needs to mount a sharepoint drive?

Other questions: 

- How does one correctly run the `onedrive` command with the docker installation?
- Which steps in [the sharepoint instructions](https://github.com/abraunegg/onedrive/blob/master/docs/SharePoint-Shared-Libraries.md) are relevant for docker?

**Comment by abraunegg** [comment, marked as answer]
@kendonB 
> Shouldn't that discourage anyone from even using docker?

Well that totally depends on how you choose to run the client - thats your choice .. you have options - Docker container or standard client.

> What is the benefit of using docker when one needs to mount a sharepoint drive?

It is your choice on how you want to run the client. Your desire might be to utilise the process isolation that Docker provides to satisfy your environment operational security requirements.

> How does one correctly run the onedrive command with the docker installation

You can login to your Docker instance and just run the 'onedrive' command - that will run with all the application defaults.

> Which steps in [the sharepoint instructions](https://github.com/abraunegg/onedrive/blob/master/docs/SharePoint-Shared-Libraries.md) are relevant for docker?

If you are wanting to configure your Docker container to connect to Sharepoint you need to:
1. Obtain your 'drive_id' for your SharePoint Library and configure this in your configuration file within your Docker instance
2. Correctly configure your Docker 'sync_dir'  so that your data is stored in a unique location

**Reply by kendonB** [reply]
Would one need a separate Docker instance for each SharePoint Library?

**Reply by abraunegg** [reply]
As per the documentation - https://github.com/abraunegg/onedrive/blob/master/docs/advanced-usage.md#configuring-the-client-to-use-multiple-onedrive-accounts--configurations - this extends to Docker

For each SharePoint Library you wish to sync, you need a separate instance. This is how the OneDrive API operates.
